src/main.h

```diff
@@ -426,6 +429,7 @@ class CScriptCheck
         std::swap(nFlags, check.nFlags);
         std::swap(cacheStore, check.cacheStore);
         std::swap(error, check.error);
+        std::swap(cachedHashesMap, check.cachedHashesMap);
```

Comment:
Ah I see. Well, I don't think we need it the swap is always called before the closure is executed (which is only when we start having concurrent access)